### PR TITLE
run more unit tests via transpiler

### PR DESCRIPTION
**Pull request by larshp**
- removes one unit test case regarding enums. Enums are not in scope, and note the code still works with enums, ref #65 

```abap
cl_abap_unit_assert=>assert_equals( exp = zif_aff_writer=>type_info-string act = cut->get_json_type_from_description( get_element_description( VALUE zif_aff_writer=>enum_type_info( ) ) ) ).
```

### CODE REVIEW

**Comment by larshp** [comment]
this runs a bunch more unit tests via transpiler, ready for review/merge

one ABAP related change, see comment above

**Comment by schneidermic0** [comment]
If we want to get rid of enum support, because AFF types shall be compatible with release 7.02, then I suggest to remove also (at least) the tested production code; `OR cl_abap_typedescr=>typekind_enum`:

https://github.com/SAP/abap-file-formats-tools/blob/4490c25515d6ea5cfbd0c37af5c52a8503be401a/src/zcl_aff_writer.clas.abap#L309

**Comment by larshp** [comment]
^can we defer that to #65 ?

its kind of a pity removing something if it works. The AFF might not be the only future use-case for this tool, so 🤷‍♂️

**Comment by schneidermic0** [comment]
I am happy if we keep the functionality, but then it would be good that it is also tested, isn't it?

I understand you removed it because the transpiler hasn't supported it yet, right? If so, wouldn't it make sense to move this single assert in a specific test method and set the test to ignore? Then we can at least run it in an ABAP system.

**Comment by larshp** [comment]
> I am happy if we keep the functionality, but then it would be good that it is also tested, isn't it?
> 
> I understand you removed it because the transpiler hasn't supported it yet, right? If so, wouldn't it make sense to move this single assert in a specific test method and set the test to ignore? Then we can at least run it in an ABAP system.

yea, good idea

**Comment by larshp** [comment]
@schneidermic0 added in above commit

File: abap_transpile.json [hunk 1]
```diff
@@ -30,110 +30,92 @@
...
-      {"object": "ZCL_AFF_WRITER", "class": "ltcl_type_writer", "method": "get_type_info_string_like"},
-      {"object": "ZCL_AFF_WRITER", "class": "ltcl_type_writer", "method": "name_mapping_and_camel_case", "note": "VALUE #( table_expr OPTIONAL )"},
-      {"object": "ZCL_AFF_WRITER", "class": "ltcl_type_writer", "method": "name_mapping_found", "note": "VALUE #( table_expr OPTIONAL )"},
-      {"object": "ZCL_AFF_WRITER", "class": "ltcl_type_writer", "method": "stack_stores_operations", "note": "VALUE #( stack[ 1 ]-operation OPTIONAL )"},
-      {"object": "ZCL_AFF_WRITER", "class": "ltcl_type_writer", "method": "validate_default", "note": "abap.types.typeTodoDecFloat16Type is not a constructor"},
-      {"object": "ZCL_AFF_WRITER", "class": "ltcl_type_writer", "method": "value_mapping_found"},
+      {"object": "ZCL_AFF_WRITER", "class": "ltcl_type_writer", "method": "stack_stores_operations", "note": "??"},
+      {"object": "ZCL_AFF_WRITER", "class": "ltcl_type_writer", "method": "get_type_info_string_like_enum", "note": "??"},
```
Comment (schneidermic0) [inline abap_transpile.json:46]: Doest it make sense to add a note here like 
```suggestion
      {"object": "ZCL_AFF_WRITER", "class": "ltcl_type_writer", "method": "get_type_info_string_like_enum", "note": "enums are not yet supported"},
```

**Review by schneidermic0 — APPROVED** [review]
Looks good to me. Just a minor question.

**Pull request merged**